src/posix/platform/spi_interface.cpp

```diff
@@ -95,9 +95,18 @@ void SpiInterface::OnRcpReset(void)
     memset(mSpiTxFrameBuffer, 0, sizeof(mSpiTxFrameBuffer));
     memset(&mInterfaceMetrics, 0, sizeof(mInterfaceMetrics));
     mInterfaceMetrics.mRcpInterfaceType = OT_POSIX_RCP_BUS_SPI;
+}
+
+otError SpiInterface::Reset(uint8_t aResetType)
+{
+    otError error = OT_ERROR_NONE;
 
+    VerifyOrExit(aResetType == Spinel::SpinelInterface::kResetHardware, error = OT_ERROR_NOT_IMPLEMENTED);
+    ResetStates();
     TriggerReset();
-    usleep(static_cast<useconds_t>(mSpiResetDelay) * kUsecPerMsec);
```

Comment:
Is this delay removed? Do we not need anymore?
 I recall this was important/useful.

Reply:
The delay is added to the `spi-hdlc-adapter` in the https://github.com/openthread/openthread/pull/3968. It is used to wait for the RCP chip starts up. Currently, after hardware resets the RCP, we are using the `WaitResponse()` to wait for the reset reason from RCP. The `WaitResponse()`  also waits for the RCP chip starts up. The timeout of the  `WaitResponse()`  is 2 seconds, it is enough for the Thread radio chip starts up.

Reply:
I just found reason (b/135504433), it used to suppress the log message `Slave did not respond to frame`. When the RCP is during the restart stage, the `INT` pin may be set to low by default, it will continuously trigger the `spi-hdlc-adapter` to start SPI transaction. Because there is no response from the RCP during the RCP restart stage, the `spi-hdlc-adapter` will output lots of log message `Slave did not respond to frame`. Adding a delay could wait for the RCP starts, and then the `INT` pin will be in a definite state, and it won't trigger invalid SPI transactions.

So we still need to keep the delay. I will restore it back.

Reply:
Nice! Thanks.

Reply:
@abtink I have updated the PR, could you help review it again?